src/averageAges.js

```diff
@@ -55,7 +72,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => {
+    const hasMather = people.some(mather => child.mother === mather.name);
+    const sexIsMale = child.sex === 'm';
```

Comment:
if you see, you duplicate `sexIsMale` and `sexIsFemale`, so you could create a function so that you filter people by certain sex